src/components/fields/location-field/location-modal/location-modal.tsx

```diff
@@ -225,7 +233,10 @@ const LocationModal = ({
 	};
 
 	const handleCloseLocationPermissionModal = () => {
-		setShowGetLocationError(false);
+		const shouldPreventDefault = !dispatchFieldEvent("click-ok-permission", id);
```

Comment:
may I suggest to rename this event to `before-hide-permission-modal`?

Reply:
Updated `click-ok-permission` to `before-hide-permission-modal`